### PR TITLE
luasocket: Replace -fpic with $(FPIC)

### PR DESCRIPTION
**Pull request by neheb**
This is causing linking errors on ppc with luasec.

Signed-off-by: Rosen Penev <rosenp@gmail.com>

Maintainer: @MikePetullo

### CODE REVIEW

File: lang/luasocket/patches/040-remove-fpic.patch [hunk 1]
```diff
@@ -0,0 +1,13 @@
+--- a/src/makefile
++++ b/src/makefile
+@@ -164,8 +164,8 @@ DEF_linux=-DLUASOCKET_$(DEBUG) -DLUA_$(COMPAT)_MODULE \
+ 	-DUNIX_API='__attribute__((visibility("default")))' \
+ 	-DMIME_API='__attribute__((visibility("default")))'
+ CFLAGS_linux= -I$(LUAINC) $(DEF) -pedantic -Wall -Wshadow -Wextra \
+-	-Wimplicit -O2 -ggdb3 -fpic -fvisibility=hidden
+-LDFLAGS_linux=-O -shared -fpic -o 
++	-Wimplicit -O2 -ggdb3 -fvisibility=hidden
```
Comment (diizzyy) [inline lang/luasocket/patches/040-remove-fpic.patch]: At least ```-Wimplicit```, ```-O2``` and ```-ggdb3``` can be removed. We shouldn't override toolchain unless there's a very good reason to do so.
```-Wall```  already sets ```-Wimplicit``` https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html

Comment (neheb) [reply lang/luasocket/patches/040-remove-fpic.patch]: I like it. Size from 52272 to 50815 on ramips.

**Comment by diizzyy** [comment]
Don't forget to update commit message accordingly to your changes

File: lang/luasocket/Makefile [hunk 2]
```diff
@@ -42,7 +42,7 @@ endef
 define Build/Compile
 	$(MAKE) -C $(PKG_BUILD_DIR)/ \
 		LIBDIR="$(TARGET_LDFLAGS)" \
-		CC="$(TARGET_CC) $(TARGET_CFLAGS) $(TARGET_CPPFLAGS) -std=gnu99" \
+		CC="$(TARGET_CC) $(TARGET_CFLAGS) $(TARGET_CPPFLAGS) $(FPIC)" \
```
Comment (karlp) [inline lang/luasocket/Makefile]: you should leave the std here, that's project code dependent, and removing it relies on the default std of our toolchains, which has changed, and might change again.

Comment (neheb) [reply lang/luasocket/Makefile]: From what I've seen, the only real impact is the change in inlining from 89 to 99. I've yet to see a difference elsewhere (not that many projects use c11 atomics).

In any case, I added -std=gnu89 and saw that it compiled fine.

**Comment by neheb** [comment]
Everything should be good to do. Merging.

**Pull request merged**